docs/root/configuration/other_features/internal_listener.rst

```diff
@@ -0,0 +1,100 @@
+
+**network.connection.client.envoy_internal** It is a client connection factory. The factory is implicitly instantiated by the dispatcher to establish a client connection to an
+internal listener address.
+
+Example config
+--------------
+Below is a smallest static config that redirect tcp_proxy on port 19000 to the tcp_proxy binding to the internal address.
+
+.. code-block:: yaml
+
+  static_resources:
+    listeners:
+    - name: outbound_tcp_svc_19000
+      address:
+        socket_address:
+          address: 0.0.0.0
+          port_value: 19000
+      filter_chains:
+      - filters:
+        - name: tcp_proxy
+          typed_config:
+            "@type": type.googleapis.com/envoy.extensions.filters.network.tcp_proxy.v3.TcpProxy
+            cluster: bridge_internal_listener
+            stat_prefix: svc_tcp_proxy
+    - name: singleton_internal_encap
+      address:
+        envoy_internal_address:
+          server_listener_name: singleton_internal_encap
+      filter_chains:
+      - filters:
+        - name: tcp_proxy
+          typed_config:
+            "@type": type.googleapis.com/envoy.extensions.filters.network.tcp_proxy.v3.TcpProxy
+            cluster: singleton_internal_encap
+            stat_prefix: encap_tcp_proxy
+    clusters:
+    - name: bridge_internal_listener
+      connect_timeout: 3600s
+      type: STATIC
+      load_assignment:
+        cluster_name: "bridge_internal_listener"
+        endpoints:
+        - lb_endpoints:
+          - endpoint:
+              address:
+                envoy_internal_address:
+                  server_listener_name: singleton_internal_encap
+      transport_socket:
+        name: envoy.transport_sockets.raw_buffer
+        typed_config:
+          "@type": type.googleapis.com/envoy.extensions.transport_sockets.raw_buffer.v3.RawBuffer
+    - name: singleton_internal_encap
+      connect_timeout: 3600s
+      type: STATIC
+      load_assignment:
+        cluster_name: "singleton_internal_encap"
+        endpoints:
+        - lb_endpoints:
+          - endpoint:
+              address:
+                socket_address:
+                  address: 0.0.0.0
+                  port_value: 19001
+  bootstrap_extensions:
+  - name: envoy.bootstrap.internal_listener
+    typed_config:
+      "@type": "type.googleapis.com/envoy.extensions.bootstrap.internal_listener.v3.InternalListener"
+  layered_runtime:
+    layers:
+    - name: enable_internal_address
+      static_layer:
+        envoy.reloadable_features.internal_address: true
+
+
+Real world use case
+-------------------
+
+**Encap http GET requests in a HTTP CONNECT request** Currently Envoy :ref:`HttpConnectionManager <envoy_v3_api_msg_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager>`
+cannot proxy a GET request in an upstream HTTP CONNECT request. This requirement
+can be acomplished by setting up the upstream endpoint of HttpConnectionManager to the internal listener address.
```

Comment:
Please link to https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/http_conn_man.html and also just spell out "HTTP connection manager". These are user rather than developer docs.